plugins/python-build/share/python-build/3.7.12

```diff
@@ -4,7 +4,7 @@ export PYTHON_BUILD_CONFIGURE_WITH_OPENSSL=1
 install_package "openssl-1.1.1k" "https://www.openssl.org/source/openssl-1.1.1k.tar.gz#892a0875b9872acd04a9fde79b1f943075d5ea162415de3047c327df33fbaee5" mac_openssl --if has_broken_mac_openssl
 install_package "readline-8.1" "https://ftpmirror.gnu.org/readline/readline-8.1.tar.gz#f8ceb4ee131e3232226a17f51b164afc46cd0b9e6cef344be87c65962cb82b02" mac_readline --if has_broken_mac_readline
 if has_tar_xz_support; then
-  install_package "Python-3.7.12" "https://www.python.org/ftp/python/3.7.12/Python-3.7.12.tar.xz#f77bf7fb47839f213e5cbf7827281078ea90de7e72b44f10d7ef385ea8c43210" standard verify_py37 copy_python_gdb ensurepip
+  install_package "Python-3.7.12" "https://www.python.org/ftp/python/3.7.12/Python-3.7.12.tar.xz#f77bf7fb47839f213e5cbf7827281078ea90de7e72b44f10d7ef385ea8c43210" arm64_apple_disable_homebrew_coreutils standard verify_py37 copy_python_gdb ensurepip
 else
-  install_package "Python-3.7.12" "https://www.python.org/ftp/python/3.7.12/Python-3.7.12.tgz#33b4daaf831be19219659466d12645f87ecec6eb21d4d9f9711018a7b66cce46" standard verify_py37 copy_python_gdb ensurepip
+  install_package "Python-3.7.12" "https://www.python.org/ftp/python/3.7.12/Python-3.7.12.tgz#33b4daaf831be19219659466d12645f87ecec6eb21d4d9f9711018a7b66cce46" arm64_apple_disable_homebrew_coreutils standard verify_py37 copy_python_gdb ensurepip
```

Comment:
This workaround only "fixes" the case where the user had added `gnubin` from Homebrew Coreutils to their PATH -- which is technically an unsupported environment.

As such, I wasn't too happy to accept it in the first place and only did it with an idea 
that it would only be applied to a few select 3.8 and 3.9 releases that do officially support Apple M1 and were only broken with GNU Coreutils because those releases sport a too old `config,sub`.

As such, do we really need it here, too?

Reply:
@pyenv/pyenv-core-maintainers Given what is said above, do you think we can drop this workaround altogether? Do we need to consider it a breaking change if yes?

Reply:
Depends on what configurations you want to support out of the box. I'm okay with whatever you prefer.

I can confirm that 3.6.15 and 3.7.12 contain the old `config.sub`, and will break with `gnubin` without the patch. My previous attempt in #2157 attempted to backport the relevant parts of `config.sub` to fix architecture detection (I wasn't aware of `arm64_apple_disable_homebrew_coreutils` until it came up in that change).

Because I depend on coreutils, my preference would be to ease building with `gnubin` in the path. I can confirm that these do build with and without `gnubin` on M1 and Intel Macs, but I can also see the reason to want to avoid any extra workarounds or configurations. (At least `arm64_apple_disable_homebrew_coreutils` is a consolidated workaround, unlike my previous approach.)

I'll go with whatever approach you prefer. I don't know how common the mix of Python 3.6/3.7, M1, and coreutils may be out there. We've hit it during setup, partly because we do development on Mac but deploy on Linux (coreutils eases compatibility concerns there). If not addressed during build, it might be worth documenting a workaround for anyone who may hit this (we'd plan to do it for our project and internal docs).

For clarification, what's considered a breaking change in this context?

Reply:
> Because I depend on coreutils

GNU Coreutils are available on PATH even without adding `gnubin`, with the `g` prefix. Whatever you depend on should be achievable by looking for an executable with `g` first, e.g.: https://github.com/pyenv/pyenv/blob/943c5f995d0ecdc957ca365aa79a9ce3294652bf/libexec/pyenv#L32

Reply:
It's more about compatibility with scripts (first-party, third-party, or customer-provided for debugging) that assume they'll be running on a typical Linux environment. Achieving a closer environment to that without modifying the scripts. Pretty important for our use cases.

Reply:
Then I wonder why you run them in an environment different than what they're designed for -- that sounds counterproductive, you're going to lose time from both Linux-aware and MacOS-aware tools breaking due to differences from either standard.

Just to let you know, we've rejected a similar workaround for GNU binutils: https://github.com/pyenv/pyenv/pull/2060#issuecomment-917723332

Reply:
I'm fine removing that workaround. I added it in based on your feedback on #2157, but it's your project, and I'll make the changes you require to get Python 3.7.12 support in in some form.

Regarding the differences in behavior, I totally understand what you mean, but this is an area that we know very well. We're old-school Linux developers who have been in this game a long time. We've built tools that were once standard on most autoconf/automake-based build systems, wrote build architecture for VMware that allowed for wide cross-Linux compatibility (along with writing most of the rest of VMware's Linux products), and helped build specs and software still in use on most Linux desktops. Though, I do agree in principal with what you're saying, and would perhaps issue the same advice to others.

But all that's orthogonal to the patch. If `arm64_apple_disable_homebrew_coreutils` usage is a blocker, I'll remove it, and we'll just document it on our end.

Reply:
With such a high-profile background, you'll surely be able to live fine without the kludge in Pyenv!
(That's what I was worried about -- if we break our users' workflow, however unsupported, we'd better provide a workable alternative for them if we'd like them to continue using our product.)

> If arm64_apple_disable_homebrew_coreutils usage is a blocker, I'll remove it,

Yes, please do.
And we can proceed to remove it from elsewhere as well.
Since it only makes a difference in an unsupported setup and we never officially documented this behavior (since it was merely a side effect of a bugfix for specific CPython releases), we don't need to consider it a breaking change. Just notifying authors of the 2.7 and coreutils' PRs should be enough (and the reverting PR will be in the next release notes).

Reply:
Sounds like a plan! I'll get these updated in the next few hours.